modules/collect/src/main/java/com/opengamma/collect/timeseries/DenseLocalDateDoubleTimeSeries.java

```diff
@@ -0,0 +1,984 @@
+   */
+  @PropertyDefinition(get = "private", validate = "notNull")
+  private final DenseTimeSeriesCalculation dateCalculation;
+
+  /**
+   * Obtains a time-series from a collection of points.
+   * The points must not contain weekend dates. If weekend
+   * dates are required then use
+   * {@link #of(Collection, DenseTimeSeriesCalculation)}.
+   *
+   * @param points  the list of points
+   * @return the time-series
+   */
+  public static DenseLocalDateDoubleTimeSeries of(Collection<LocalDateDoublePoint> points) {
+    return of(points, SKIP_WEEKENDS);
+  }
+
+  /**
+   * Obtains a time-series from a collection of points
+   * using the specified {@code dateCalculation}.
+   *
+   * @param points  the list of points
+   * @return the time-series
+   */
+  public static DenseLocalDateDoubleTimeSeries of(
+      Collection<LocalDateDoublePoint> points,
+      DenseTimeSeriesCalculation dateCalculation) {
+
+    ArgChecker.notNull(points, "points");
+
+    ArgChecker.isFalse(
+        points.stream().anyMatch(pt ->
+            pt == null || (dateCalculation == SKIP_WEEKENDS && isWeekend(pt.getDate()))),
+        "dates must not contain nulls or weekend dates");
+
+    if (points.isEmpty()) {
+      return EMPTY_SERIES;
+    }
+
+    Pair<LocalDate, LocalDate> extremes = points.stream()
+        .reduce(Pair.of(LocalDate.MAX, LocalDate.MIN), (pair, pt) -> {
+          LocalDate earliest = pair.getFirst();
+          LocalDate latest = pair.getSecond();
+          LocalDate ld = pt.getDate();
+          return Pair.of(ld.isBefore(earliest) ? ld : earliest, ld.isAfter(latest) ? ld : latest);
+        }, (p1, p2) -> p1);
+
+    LocalDate startDate = extremes.getFirst();
+
+    int size = dateCalculation.calculatePosition(startDate, extremes.getSecond()) + 1;
+    double[] values = new double[size];
+    if (points.size() != size) {
+      // We have gaps - prefill with NaNs
+      Arrays.fill(values, Double.NaN);
+    }
+
+    points.stream()
+        .forEach(p -> values[dateCalculation.calculatePosition(startDate, p.getDate())] = p.getValue());
+
+    return new DenseLocalDateDoubleTimeSeries(startDate, values, dateCalculation);
+  }
+
+  /**
+   * Obtains a time-series from a map of dates and values.
+   * The points must not contain weekend dates. If weekend
+   * dates are required then use
+   * {@link #of(Map, DenseTimeSeriesCalculation)}.
+   *
+   * @param map  the map of dates to values
+   * @return the time-series
+   */
+  public static DenseLocalDateDoubleTimeSeries of(Map<LocalDate, Double> map) {
+    return of(map, SKIP_WEEKENDS);
+  }
+
+  /**
+   * Obtains a time-series from a map of dates and values
+   * using the specified {@code dateCalculation}.
+   *
+   * @param map  the map of dates to values
+   * @return the time-series
+   */
+  public static DenseLocalDateDoubleTimeSeries of(
+      Map<LocalDate, Double> map,
+      DenseTimeSeriesCalculation dateCalculation) {
+
+    // TODO handle empty map
```

Comment:
Is this comment obsolete now? There's an early return below that checks if the map is empty.